Swiftfin/Components/FilterDrawerHStack/FilterDrawerHStack.swift

```diff
@@ -85,4 +61,19 @@ extension FilterDrawerHStack {
     func onSelect(_ action: @escaping (FilterCoordinator.Parameters) -> Void) -> Self {
         copy(modifying: \.onSelect, with: action)
     }
+    
+    // Finds the Filters/Genres/SortBy/Etc Property from the the ItemsFilter & 
+    func getFilterProperty<T>(from object: T, propertyName: String) -> [ItemFilters.Filter] {
```

Comment:
One should rarely have to use reflection, if at all. There is already the `ItemFilters` key path defined on the `enum`, we need to use that instead.

Reply:
Gotcha! I was worried the nesting looping was a bit over-engineered for this (plus a bit inefficient). Just for future reference, why is reflection a rarely-or-never function? I'm still pretty new to Swift.

Since I was calling this from viewModel.currentFilters I wasn't sure what was the best way to get this call done. Below is the reworked method that I put on the FilterDrawerActionButton enum instead. Sorry for the screenshot I wasn't sure how to reference the code directly. Is there a better way I should be making this call?

<details>
<summary>Re-Worked isItemFilterActive</summary>

<img src="https://github.com/jellyfin/Swiftfin/assets/37599297/e7b3da6b-4edd-4ccb-b311-1df2d331d2fd" width="30%" height="30%">
</details>

<details>
<summary>Re-Worked FilterDrawerHStack ForEach</summary>

<img src="https://github.com/jellyfin/Swiftfin/assets/37599297/1ac610ff-cae2-465a-8a74-4df182597f92" width="30%" height="30%">
</details>

Reply:
Moved Filter == Activated test to the enum. See previous comment with screenshot. This should be in the PR branch now.

Reply:
Er saying "if at all" is a bit dogmatic from me, but reflection just isn't necessary and is used in very purposeful cases. That just happens to be very rare.